src/java/org/apache/cassandra/config/Converters.java

```diff
@@ -0,0 +1,122 @@
+                     o -> ((DurationSpec)o).toMinutes()),
+    MEBIBYTES_DATA_STORAGE(Long.class,
+                          o -> DataStorageSpec.inMebibytes((Long) o),
+                          o -> ((DataStorageSpec)o).toMebibytes()),
+    KIBIBYTES_DATASTORAGE(Long.class,
+                          o -> DataStorageSpec.inKibibytes((Long) o),
+                          o -> ((DataStorageSpec)o).toKibibytes()),
+    BYTES_DATASTORAGE(Long.class,
+                      o -> DataStorageSpec.inBytes((Long) o),
+                      o -> ((DataStorageSpec)o).toBytes()),
+    MEBIBYTES_PER_SECOND_DATA_RATE(Long.class,
+                                   o -> DataRateSpec.inMebibytesPerSecond((Long) o),
+                                   o -> ((DataRateSpec)o).toMebibytesPerSecond()),
+    /**
+     * This converter is a custom one to support backward compatibility for stream_throughput_outbound and
+     * inter_dc_stream_throughput_outbound which were provided in megatibs per second prior CASSANDRA-15234.
+     */
+    MEGABITS_TO_MEBIBYTES_PER_SECOND_DATA_RATE(Long.class,
+                                          o -> DataRateSpec.megabitsPerSecondInMebibytesPerSecond((Long)o),
+                                          o -> ((DataRateSpec)o).toMegabitsPerSecond());
+
+    private final Class<?> inputType;
+    private final Function<Object, Object> convert;
+    private final Function<Object, Object> reverseConvert;
+
+    Converters(Class<?> inputType, Function<Object, Object> convert, Function<Object, Object> reverseConvert)
+    {
+        this.inputType = inputType;
+        this.convert = convert;
+        this.reverseConvert = reverseConvert;
+    }
+
+    /**
+     * A method to identify what type is needed to be returned by the converter used for a configuration parameter
+     * in {@link Replaces} annotation in {@link Config}
+     * @return class type
+     */
+    public Class<?> getInputType()
+    {
+        return inputType;
+    }
+
+    /**
+     * Apply the converter specified as part of the {@link Replaces} annotation in {@link Config}
+     * @param value we will use from cassandra.yaml to create a new {@link Config} parameter of type {@link DurationSpec},
+     * {@link DataRateSpec} or {@link DataStorageSpec}
+     * @return new object of type {@link DurationSpec}, {@link DataRateSpec} or {@link DataStorageSpec}
+     */
+    public Object apply(Object value)
+    {
+        if (value == null) return null;
+        return convert.apply(value);
+    }
+
+    /**
+     * Apply the converter specified as part of the {@link Replaces} annotation in {@link Config} to get config parameters'
+     * values in the old format pre-CASSANDRA-15234 and in the right units, used in the Virtual Tables to ensure backward
+     * compatibility
+     * @param value we will use to calculate the output value
+     * @return the numeric value
+     */
+    public Object reverseApply(Object value)
```

Comment:
nit/naming: We could also try `toNewValue()` and `toOldValue()` or `convert()` and `deconvert()` (although YAML properties don't have religion https://www.yourdictionary.com/deconvert)

Reply:
I like the `convert`/`deconvert` idea